packages/backend-core/src/db/lucene.ts

```diff
@@ -463,6 +474,13 @@ export class QueryBuilder<T> {
       allOr = false
       build({ tableId }, equal)
     }
+    if (allFiltersEmpty) {
+      if (this.#query.onEmptyFilter === EmptyFilterOption.RETURN_NONE) {
```

Comment:
Nice work making sure that we're comparing to `NONE`, since that will account for all the existing saved filters which have an `undefined` `onEmptyFilter` filter, meaning we'll keep the same value as before. Would be easy to overlook this!